grc/core/blocks/block.py

```diff
@@ -11,15 +11,15 @@
 import collections
 import itertools
 import copy
-
 import re
-
 import ast
+import typing
```

Comment:
Type annotations can be restricted to "not runtime" with something like:

```python
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Optional, # ... etc
```

Reply:
hm, what's the advantage of that? I tried that in a place where annotating with a more detailed type than `Element` was not possible due to circular imports, and it still falls on my feet during precompilation.

Reply:
Even with `__future__.annotations`? Ultimately I think preventing the import of things only used for typing can help with runtime speed esp when actually running the GUI (boot times). Of course that might be miniscule and might be premature optimization, though I do also find it personally cleaner to do these typing guards to restrict all typing related code to one place. Of course that could just be personal taste!

Reply:
ah, I didn't do that, as afaik that's Py 3.7, and we still are only depending on Py 3.6.5 …

We should probably raise that for `main`, but we can't for `maint-3.10`…

Reply:
3.6?? Even 3.7 is end-of-life with 3.8 EOL coming next October....oof

Reply:
Ubuntu 20.04 has Python 3.8, so I think we can relax the version requirement. The Python 3.6 requirement was based on Ubuntu 18.04. We were trying to keep GR 3.9 and GR 3.10 compatible at the time.

@mormj do you remember if we are intentionally allowing Python 3.6? @balister are there embedded platforms running GR 3.10 but Python 3.6?

Reply:
Hmm it would be interesting to see how much time the type annotations add to the GRC startup time

Reply:
I'm supposing there could be some people still using RHEL type distros unfortunately - but I haven't seen 3.10 running on them anyhow.  I think the 3.6 decision was made a LONG time ago (like GR3.7-->3.8), and should be reevaulated.

Reply:
I *think* the decision for 3.6.5 went like 

> OK, how do we break portability to the least platforms, but directly end up on a version of Python3 that fully supports all the `f"An error occurred ({error}), ejecting the humans to space. Sorry Dave."` strings we want

and then we as @willcode said, there was motivation for bringing GR 3.9 to Ubuntu 18.04